releasenotes/notes/33293.yaml

```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: security
+issue:
+  - 33293
+releaseNotes:
+  - |
+    **Fixed** a bug where secret files are not watched once they are removed and added back.
```

Comment:
would be great to add a test to TestAgent, I think we have the setup require to test it easily

Reply:
I will add a test